src/storage/ExtensionBasedMapper.ts

```diff
@@ -0,0 +1,129 @@
+import { posix } from 'path';
+import { types } from 'mime-types';
+import { RuntimeConfig } from '../init/RuntimeConfig';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { APPLICATION_OCTET_STREAM, TEXT_TURTLE } from '../util/ContentTypes';
+import { ConflictHttpError } from '../util/errors/ConflictHttpError';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { trimTrailingSlashes } from '../util/Util';
+import { FileIdentifierMapper } from './FileIdentifierMapper';
+
+const { join: joinPath, normalize: normalizePath } = posix;
+
+export interface PathAbstraction {
+
+  /**
+   * The parent directory path of the file.
+   */
+  path: string;
+
+  /**
+   * The name/slug of the file.
+   */
+  slug?: string;
```

Comment:
Slug or name? And what if there is no slug, is there no name then?

Reply:
Seems like I was kind of confused on what slug is here, as it seems like the function splits the path like directory+filename. But it is actually coming from the requested URL, so I will just remove 'name' from the comments.

Reply:
So it's very likely _not_ a slug then, because a slug is a naming suggestion. It's likely `name` then. (But hopefully not optional?)

Reply:
It's a bit weird, it helps to look at where it was originally ([link](https://github.com/solid/community-server/pull/142/files#diff-47500dd20ac8cef4ed6f04efea1e4052R148-R161). It extracts the name part right of the last slash to see if the incoming PUT wants to create/update a file resource or a container. In the cases where it is a container the "slug" part will be empty, and this will be used by the interaction controller to check if the URI corresponds to a container or resource.

Reply:
So it's `containerPath` and `documentName` then?

Reply:
Yea that would make sense.

Reply:
@freyavs would also rename the function to something else than `extractSlug` then. Perhaps `getResourcePath` since that is what it does now.